sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/predicates.scala

```diff
@@ -236,24 +236,34 @@ case class In(value: Expression, list: Seq[Expression]) extends Predicate {
   override def doGenCode(ctx: CodegenContext, ev: ExprCode): ExprCode = {
     val valueGen = value.genCode(ctx)
     val listGen = list.map(_.genCode(ctx))
+    ctx.addMutableState("boolean", ev.value, "")
+    ctx.addMutableState("boolean", ev.isNull, "")
```

Comment:
so we will create 2 mutable stats anyway?

Reply:
Yes if we call `ctx.splitExpressions`. One is to keep a value, the other is to keep nullability. 
Is it better to avoid to create 2 mutable if we will not call `ctx.splitExpressions` (i.e. `!(ctx.INPUT_ROW != null && ctx.currentVars == null)))?